adapters/oath/oath.go

```diff
@@ -0,0 +1,178 @@
+package oath
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"strconv"
+)
+
+//bid URL
+const uri = "http://east-bid.ybp.yahoo.com/bid/appnexuspbs"
+
+type OathAdapter struct {
+	http *adapters.HTTPAdapter
+	URI  string
+}
+
+func (a *OathAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+
+	if len(request.Imp) == 0 {
+		err := &adapters.BadInputError{
+			Message: fmt.Sprintf("No impression in the bid request"),
+		}
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	var bannerImps []openrtb.Imp
+	var videoImps []openrtb.Imp
+
+	for _, imp := range request.Imp {
```

Comment:
It seems the purpose of this loop is to simply validate that at least one valid Imp exists. May be more efficient to simply declare `validImps := false`, and then when you find a banner or video imp, set `validImps = true` rather than building these two lists that you will never actually use. Avoiding the allocation extra memory as the lists grows may save a milisecond that can be spent waiting for the response before the timeout hits. :)